drivers/spi/spi_nrfx_spi.c

```diff
@@ -270,24 +270,12 @@ static int init_spi(struct device *dev, const nrfx_spi_config_t *config)
 	return 0;
 }
 
-static void spi_isr(void *irq_handler)
-{
-	((nrfx_irq_handler_t)irq_handler)();
-}
-
-/* In Nordic SoCs the IRQ number assigned to a peripheral is equal to the ID
- * of the block of 0x1000 bytes in the peripheral address space assigned to
- * this peripheral. See the chapter "Peripheral interface" (sections "Peripheral
- * ID" and "Interrupts") in the product specification of a given SoC.
- */
-#define NRF_SPI_IRQ_NUMBER(idx)  (uint8_t)((uint32_t)NRF_SPI##idx >> 12u)
-
 #define SPI_NRFX_SPI_DEVICE(idx)					\
 	static int spi_##idx##_init(struct device *dev)			\
 	{								\
-		IRQ_CONNECT(NRF_SPI_IRQ_NUMBER(idx),			\
+		IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_SPI##idx),		\
```

Comment:
we should get irq number from device tree.

Reply:
There is no DTS support for nRF SPI peripherals yet. I'll add it in a next step. Or do you want me to do it in this PR?

Reply:
we can do this in a followup PR.